.github/workflows/buildrepo.yaml

```diff
@@ -5,8 +5,25 @@ on:
   push:
     branches: [ "master" ]
 
+  workflow_dispatch:
+    inputs:
+      skip_upload:
+        required: false
+        description: "Skip upload"
+        type: boolean
+        default: false
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
```

Comment:
What is the permission about? Why do we need write access?

Reply:
Something about fetching a token which requires write